src/core/sys/posix/sys/time.d

```diff
@@ -28,7 +28,55 @@ else version (WatchOS)
     version = Darwin;
 
 version (Posix):
-extern (C) nothrow @nogc:
+nothrow @nogc:
+
+
+/* macros provided to operate on timeval structures
+ * they are extern(D) because they are not really C symbols, just macros
+ */
+
+pure @safe void timeradd(const timeval* a, const timeval* b,
+        timeval* result)
+{
+    result.tv_sec = a.tv_sec + b.tv_sec;
+    result.tv_usec = a.tv_usec + b.tv_usec;
```

Comment:
I think this, being an interface to C code, just have to do the exact same as C does. Sometimes is not just a matter of porting C code that then will be just "pure D", sometimes, at least at Sociomantic, we just prefer to use C functions because they are proven and we know what to expect. OK, this is a grey zone because the actual code will be generated by the D compiler, not linked, but having the exact same implementation as in C has some value IMHO.

Remember these are just plain C bindings, not phobos, things here just have to be as close to C as possible, not be D idioms to achieve the same you could achieve using the C counterparts but more conveniently and safely.